🚀 2단계 - 요구 사항 정리

MoonHKLee opened the pull request:
잘 부탁드립니다.

MoonHKLee committed:
docs(readme) : README 업데이트 STEP 2 세부 내용 추가

README.md 파일의 STEP 2 섹션에 여러 비즈니스 요구사항 정의가 추가되었습니다. 메뉴, 메뉴 그룹, 주문, 주문 테이블, 상품 등 다양한 도메인에 대한 요구사항이 기술되어 있습니다. 각 도메인에 대한 요구사항은 구체적인 행동과 그에 따른 제약사항을 포함하고 있습니다.

kyucumber commented on README.md:
```diff
@@ -12,24 +12,119 @@ docker compose -p kitchenpos up -d
+    - 주문 타입이 매장 내 식사라면 테이블 구분자가 필요하다.
+      - 테이블은 등록된 테이블이어야한다.
+      - 테이블에 앉아있는 상태여야한다.
+  - 주문을 승인한다.
```
각 주문 항목 별 요구사항을 상세하게 잘 정리해주셨네요. 👍

다만 현재 항목별 주문 순서가 한눈에 보이지는 않는데, 요구사항 명세를 통해 도메인을 이해하는데 도움이 되는 경우도 있으므로 아래와 같이 주문 타입 별 상태가 어떻게 흘러가는지에 대해 정리해보아도 좋을 것 같네요.

- 배달 주문
  - 주문 승인 -> 배달 시작 -> 배달 완료

MoonHKLee replied:
넵 감사합니다.

kyucumber commented on README.md:
```diff
@@ -12,24 +12,119 @@ docker compose -p kitchenpos up -d
+
+- 메뉴
+  - 메뉴를 만든다.
+    - 가격은 0 이상이어야 한다.
```
중요한 부분은 아니지만 단위도 적어주시면 좋을 것 같아요!

```suggestion
    - 가격은 0원 이상이어야 한다.
```

kyucumber commented on README.md:
```diff
@@ -12,24 +12,119 @@ docker compose -p kitchenpos up -d
+    - 메뉴의 가격과 메뉴의 상품의 가격 총합이 같아야한다.
+  - 메뉴를 노출한다.
+    - 등록된 메뉴만 보여줄 수 있다.
+    - 메뉴의 가격과 메뉴의 상품의 가격 총합이 같아야한다.
```
같아야 하는 조건이 아니었던 것 같아요. 다시 한번 확인해보고 요구사항을 정리해보시면 좋을 것 같네요!

MoonHKLee replied:
같거나 작아야 한다였네요!

kyucumber reviewed (changes requested):
요구사항을 아주 꼼꼼하게 잘 정리해주신 것 같습니다.

간단한 코멘트 남겨두었으니 확인 부탁드려요. 다음 요청 시 머지하도록 할게요.  🙇

MoonHKLee committed:
docs : 요구사항 갱신

가격 단위 추가.

Co-authored-by: KyuNam <tramyu@naver.com>

MoonHKLee committed:
docs (readme): 메뉴 가격 기준 수정 및 주문 흐름도 추가

README 파일에서 메뉴 가격의 기준이 수정되었습니다. 이전의 '메뉴의 가격과 상품의 가격 총합이 같아야 한다'는 조건에서 '메뉴의 가격과 상품의 가격 총합이 같거나 작아야 한다'로 변경되었습니다. 또한, 주문 기능 설명에 주문 흐름도 이미지가 추가되었습니다.

kyucumber reviewed (approved):
고생하셨습니다.

다음 단계 진행을 위해 머지할게요.
